### PR TITLE
Task 2: Data types

### PR DESCRIPTION
**Pull request by osechka333**

### CODE REVIEW

File: Exercises/data-types-overview/data-types-overview.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+//1. Declare a variable on each type using two options: plain creation, creation with constructor.
+let numberVar = 32;
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js]: Better always use  `const` if you are not reassign variable

Comment (lizarusi) [reply Exercises/data-types-overview/data-types-overview.js]: also you missed creation with constructor

File: Exercises/data-types-overview/data-types-overview.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+console.log(typeof stringVar);
+console.log(typeof booleanVar);
+console.log(typeof nullVar); 
+console.log(typeof undefinedVar);
+console.log(typeof objectVar);
+console.log(typeof bigIntVar);
+
+//3. Demonstrate object's references.
+
+objectValue = { age: 32, position: 'product owner', priorExperience: true };
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js]: keyword `const/let` missed here `objectValue` will appear in global object which is not good.

File: Exercises/data-types-overview/data-types-overview.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+objectValue.age = 34;
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js]: by object references meant to demonstrate that variable contains just reference to object in memory and when you assign this variable to other variable they will affect each other on change.

File: Exercises/data-types-overview/data-types-overview.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+//5. Declare an array containing different data types.
+
+let candidate = [ age = 32, position = 'product owner', priorExperience = true, gender = 'male', illness = null, bankAccount = BigInt(9007199254740991)]; 
+
+console.log(candidate);
+
+//6. Iterate through it and collect amount of each data type to object.
+
+candidate.forEach(function(element) {
+  console.log(typeof element);
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js]: not actually a collecting, I want you to collect how much times some type appear in array

**Review by lizarusi — CHANGES_REQUESTED** [review]

File: Exercises/data-types-overview/data-types-overview.js (hunk 1 above)
Comment (osechka333) [reply Exercises/data-types-overview/data-types-overview.js]: thanks! fixed

File: Exercises/data-types-overview/data-types-overview.js (hunk 4 above)
Comment (osechka333) [reply Exercises/data-types-overview/data-types-overview.js]: I made the fix with Object.keys but I am not sure that is the best way

candidate.forEach(function(item) {
  console.log(typeof item); 
  console.log(Object.keys(item).lenght);
});

File: Exercises/data-types-overview/data-types-overview.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+console.log(objectValue);
+console.log(anotherObjectValue);
+
+//4. Demonstrate string's backticks features.
+
+console.log(`${stringVar} is ${numberVar} years old and his CPR number is ${bigIntVar}`);
+
+//5. Declare an array containing different data types.
+
+const candidate = [ age = 32, position = 'product owner', priorExperience = true, gender = 'male', bankAccount = BigInt(9007199254740991)];
```
Comment (Mike-Savin) [inline Exercises/data-types-overview/data-types-overview.js]: Actually such syntax (I mean [name1 = value1, name2 = value2] ) is not a correct way to declare an array.
Array is not a set of key-value pairs, it is a set of values only.
What you did was declaring global variables "age", 
"position", "priorExperience" etc. and adding their values to the array.

File: Exercises/data-types-overview/data-types-overview.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+candidate.forEach(function(item) {
+  console.log(typeof item); 
+  console.log(Object.keys(item).lenght);
+});
+
+const array = [120, 45, 66, 88, true, false],
+  types = {};
+
+array.forEach(v => types[typeof v] = true);
+
```
Comment (Mike-Savin) [inline Exercises/data-types-overview/data-types-overview.js]: This solution is almost right but you need to not just mark some type presence as "true" - you need to save the number of such type occurrences in the array. Like the result should be:
types = {number: 4, boolean: 2};

**Review by Mike-Savin — CHANGES_REQUESTED** [review]

File: Exercises/data-types-overview/data-types-overview.js (hunk 5 above)
Comment (osechka333) [reply Exercises/data-types-overview/data-types-overview.js]: thanks! Got the point and made the fix

File: Exercises/data-types-overview/data-types-overview.js (hunk 6 above)
Comment (osechka333) [reply Exercises/data-types-overview/data-types-overview.js]: thanks for the help!

File: Exercises/data-types-overview/data-types-overview.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
+//1. Declare a variable on each type using two options: plain creation, creation with constructor.
+const numberVar = 32;
+const stringVar = 'Product Owner';
+const booleanVar = true;
+const nullVar = null;
+let undefinedVar; 
+const objectVar = { a: 'prior experience', b: 'resume'};
+const bigIntVar = BigInt(9007199254740991);
+
+class User {
```
Comment (Mike-Savin) [inline Exercises/data-types-overview/data-types-overview.js:10]: We meant other and simpler thing by creation with constructor.

For example, you can create a number variable like this:
```js
let num = new Number(4);
```
And similarly you can declare string, array or object variables.
Such syntax is not recommended to use, but it's good to know about it and about the difference in typeof behavior on usual variables and variables defined by such constructors.

**Review by Mike-Savin — CHANGES_REQUESTED** [review]
Looks good, just one little thing to fix

File: Exercises/data-types-overview/data-types-overview.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+const candidate = [ 32, 'product owner', true, 'male', BigInt(9007199254740991)]; 
+console.log(candidate);
+
+//6. Iterate through it and collect amount of each data type to object.
+
+const array = [120, 45, 66, 88, true, false];
+const types = {};
+
+array.forEach(v => {
+
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js]: odd space, don't forget to run `npm run lint`

**Review by lizarusi — CHANGES_REQUESTED** [review]
see @Mike-Savin comment, anyway great job!

File: Exercises/data-types-overview/data-types-overview.js (hunk 7 above)
Comment (osechka333) [reply Exercises/data-types-overview/data-types-overview.js:10]: thanks for the comment! Reworked two variables:
const numberVar = new Number(32);
const objectVar = new Object('prior experience');

File: Exercises/data-types-overview/data-types-overview.js (hunk 8 above)
Comment (osechka333) [reply Exercises/data-types-overview/data-types-overview.js]: I run it and got no errors about this additional space! 
Anyway thanks. I removed it

File: Exercises/data-types-overview/data-types-overview.js [hunk 9]
```diff
@@ -0,0 +1,56 @@
+//1. Declare a variable on each type using two options: plain creation, creation with constructor.
+const stringVar = 'Product Owner';
+const booleanVar = true;
+const nullVar = null;
+let undefinedVar; 
+const numberVar = new Number(32);
+const objectVar = new Object('prior experience');
```
Comment (Mike-Savin) [inline Exercises/data-types-overview/data-types-overview.js:7]: You can also check this comment to get some information about declaring numbers/strings/booleans using `new` operator:
https://stackoverflow.com/a/4719334

**Review by Mike-Savin — APPROVED** [review]
Looks good

**Review by lizarusi — APPROVED** [review]
Cool!

**Pull request merged**